### PR TITLE
[Feature] add option to purge created Virtualenvs

### PR DESCRIPTION
**Pull request by jamestansx**
Hi, the rationale behind this PR are:

1. to be able to remove Virtualenvs that the associated projects are deleted.
2. to be able to remove all Virtualenvs at once for cleanup purpose

This implementation might not be neat, please review it. Thanks!

### CODE REVIEW

File: src/pdm_venv/utils.py [hunk 1]
```diff
@@ -39,3 +39,11 @@ def get_venv_python(venv: Path) -> Path:
     """Get the interpreter path inside the given venv."""
     suffix = ".exe" if IS_WIN else ""
     return venv / BIN_DIR / f"python{suffix}"
+
+
+def get_all_venvs(project: Project) -> Iterable[Tuple[str, Path]]:
```
Comment (frostming) [inline src/pdm_venv/utils.py:44]: why do you create another function to show the path prefix in the venv name?

**Comment by jamestansx** [comment]
I have added `--interactive` mode to allow users to select particular venv to be removed. 

---

The difference between `purge` and `remove` is:

- The scope of `remove` is local, while `purge` is global

This is to give option to user for maintain the `venv` created (for disk space or for cleanup).

**Comment by frostming** [comment]
Mostly good, a few test cases to verify the change are needed before merging.

**Comment by jamestansx** [comment]
I have added the test, please review it. Thanks.

File: tests/test_commands.py [hunk 2]
```diff
@@ -75,3 +77,40 @@ def test_venv_auto_create(invoke, mocker):
...
+    assert result.exit_code == 0, result.stderr
+    venv_path = re.match(
+        r"Virtualenv (.+) is created successfully", result.output
+    ).group(1)
+    result = invoke(["venv", "purge", "-f"])
+    assert result.exit_code == 0, result.stderr
+    assert not os.path.exists(venv_path)
+
+
+user_options = [("none", 1), ("0", 0), ("all", 0)]
```
Comment (frostming) [inline tests/test_commands.py:104]:
```suggestion
user_options = [("none", True), ("0", False), ("all", False)]
```

**Comment by frostming** [comment]
LGTM, thanks for your effort

**Pull request merged**